middleware/requestID.go

```diff
@@ -68,6 +67,19 @@ func RequestID(next http.Handler) http.Handler {
 	})
 }
 
+// RequestIDWithCustomHeaderStrKey is a middleware that injects a request ID into the context of each
+// request. Different from RequestID, this middleware uses a custom header key to get the request ID,
+// and does not generate a new request ID if the custom header key is not present.
+func RequestIDWithCustomHeaderStrKey(headerStrKey string) func(next http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			requestID := req.Header.Get(headerStrKey)
```

Comment:
@wmattei 

In this particular line, If the client has _not_ set the Request ID header OR for an empty Request ID for that particular request, the `requestID` here will be an empty quote "". Having worked with only server generated request IDs, I'm a bit curious about the generally accepted standards here - 

- Should the server revert with a `400` as the client has forgotten to set the header? 
- What about if the Header is present but is empty?